src/ceylon/language/List.ceylon

```diff
@@ -851,13 +851,12 @@ shared interface List<out Element=Anything>
         clone() => outer.clone().Rest(from);
         
         iterator() 
-                => let (iter = outer.iterator()) 
+                => let (o = outer)
             object satisfies Iterator<Element> {
                 variable value i=0;
-                while (i++<from) {
-                    iter.next();
-                }
-                next() => iter.next();
+                next() => if (i < outer.size)
```

Comment:
@sadmac7000 that doesn't look quite right. Shouldn't it be `if (i < outer.size-from)`, or, even better, just start at `value i = from`?